src/main/java/duke/CommandParser.java

```diff
@@ -0,0 +1,74 @@
+package duke;
+
+/**
+ * A class representing a parser for commands from the CLI.
+ */
+public class CommandParser extends Parser<String[]> {
+	private final static String[] COMMANDS_WITH_ARGS = new String[] {"todo", "deadline", 
+			"event", "delete", "done", "find"};
+	private final static String[] COMMANDS_WITHOUT_ARGS = new String[] {"list", "bye"};
+	private final static String UNKNOWN_COMMAND_MSG = "☹ OOPS!!! " +
+			"I'm sorry, but I don't know what that means :-(";
+	private static final String MISSING_ARGUMENT_TEMPLATE = "☹ OOPS!!! " +
+			"Seems like there are missing argument(s) for %s";
+	private static final String NON_NUMBER_ARGUMENT_TEMPLATE = "☹ OOPS!!! " +
+			"Should have entered a number for %s, BOI";
+
+	/**
+	 * Parse the given line from the CLI and returns an array of commands and arguments.
+	 * 
+	 * @param cmd Line from the CLI.
+	 * @return An array representing the command and arguments.
+	 * @throws DukeException If the command is invalid or has invalid arguments.
+	 */
+	public String[] parse(String cmd) throws DukeException {
+		cmd = cmd.trim();
+		boolean isInvalidCommand = true;
+
+		String[] cmdSplit = cmd.split("[ \\t]+", 2);
+		for (String cmdCheck : COMMANDS_WITHOUT_ARGS) {
+			if (cmd.equals(cmdCheck)) {
+				return new String[]{cmd};
+			} else if (cmdSplit[0].equals(cmdCheck)) {
+				throw new DukeException("☹ OOPS!!! Unknown Argument for " + cmdCheck);
+			}
+		}
+
+		for (String cmdCheck : COMMANDS_WITH_ARGS) {
+			isInvalidCommand = isInvalidCommand && !(cmdSplit[0].equals(cmdCheck));
```

Comment:
Should we assign a variable to define a meaning of `!(cmdSplit[0].equals(cmdCheck))` ?

Reply:
I see, in my opinion its more like the same level of abstraction, if the left side is `isInvalidCommand`, the right side usually is not supposed to do some lower-level operations, it just makes it easier to read